Realtek rtl8723bs chip driver has a bug in AP mode
When client connects to the access point the driver causes
kernel panic.
This patch is from Armbian project fixes this behavior.

Tested on: Xiaomi Gateway DGNWG05LM

Signed-off-by: Ivan Belokobylskiy <belokobylskij@gmail.com>

--- a/drivers/staging/rtl8723bs/os_dep/ioctl_cfg80211.c
+++ b/drivers/staging/rtl8723bs/os_dep/ioctl_cfg80211.c
@@ -2424,6 +2424,7 @@ void rtw_cfg80211_indicate_sta_assoc(struct adapter *padapter, u8 *pmgmt_frame,
 			ie_offset = _REASOCREQ_IE_OFFSET_;

 		sinfo.filled = 0;
+		sinfo.pertid = 0;
 		sinfo.assoc_req_ies = pmgmt_frame + WLAN_HDR_A3_LEN + ie_offset;
 		sinfo.assoc_req_ies_len = frame_len - WLAN_HDR_A3_LEN - ie_offset;
 		cfg80211_new_sta(ndev, GetAddr2Ptr(pmgmt_frame), &sinfo, GFP_ATOMIC);
